src/github.com/getlantern/balancer/strategy.go

```diff
@@ -0,0 +1,106 @@
+package balancer
+
+import (
+	"math/rand"
+)
+
+// Random strategy gives even chance to each dialer, act as a baseline to other
+// strategies.
+func Random(dialers []*dialer) dialerHeap {
+	return dialerHeap{dialers, func(i, j int) bool {
+		// we don't need good randomness, skip seeding
+		if rand.Intn(2) == 0 {
+			return false
+		}
+		return true
+	}}
+}
+
+// Sticky strategy always pick the dialer with largest consecutive success
+// count or the smallest consecutive failure count
+func Sticky(dialers []*dialer) dialerHeap {
+	return dialerHeap{dialers, func(i, j int) bool {
+		mi := dialers[i].metrics()
+		mj := dialers[j].metrics()
+		return (mi.consecSuccesses - mi.consecFailures) >
+			(mj.consecSuccesses - mj.consecFailures)
+	}}
+}
+
+// Fastest strategy always pick the dialer with lowest recent average connect time
+func Fastest(dialers []*dialer) dialerHeap {
+	return dialerHeap{dialers, func(i, j int) bool {
+		mi := dialers[i].metrics()
+		mj := dialers[j].metrics()
+		return mi.avgConnTime < mj.avgConnTime
```

Comment:
See [pinger.go](https://github.com/getlantern/http-proxy-lantern/blob/master/pinger/pinger.go) for code that pings the proxy server.